docs/user/tutorial-websockets.rst

```diff
@@ -0,0 +1,608 @@
+.. _tutorial-websockets:
+
+Tutorial (WebSockets)
+=====================
+
+In this tutorial, we're going to build a WebSocket server using Falcon. We'll start with a simple server that echoes back any message it receives.
+We'll then add more functionality to the server, such as sending JSON data and logging messages.
+
+.. note::
+   This tutorial covers the asynchronous flavor of Falcon using
+   the `ASGI <https://asgi.readthedocs.io/en/latest/>`__ protocol.
+
+   A Falcon WebSocket server is build upon the `ASGI WebSocket specification <https://asgi.readthedocs.io/en/latest/specs/www.html#websocket>`__. Therefore it's not supported in a Falcon WSGI application.
+
+
+First Steps
+___________
+
+We'll start with a clean working directory and create a new virtual environment using the `venv` module.:
+
+.. code-block:: bash
+
+   $ mkdir asyncws
+   $ cd asyncws
+   $ python3 -m venv .venv
+   $ source .venv/bin/activate
+
+Create the following directory structure::
+
+  asyncws
+  ├── .venv
+  └── asyncws
+      ├── __init__.py
+      └── app.py
+
+
+And next we'll install Falcon and Uvicorn in our freshly created virtual environment:
+
+.. code-block:: bash
+
+   $ pip install falcon uvicorn
```

Comment:
I think you need to install `uvicorn[standard]`, see also [here](https://www.uvicorn.org/#quickstart) in order to get WebSocket support. You could also link to our installation docs.

**Edit**: Ah, I see you ask the user to install it later. :)

Reply:
Yep, I've did that on purpose to make it clear it's a dependency. Just in case someone doesn't use the `standard` extra :)